src/netgearpy/models.py

```diff
@@ -23,3 +26,48 @@ def __pre_deserialize__(cls, d: dict[str, Any]) -> dict[str, Any]:
             **d,
             "LoginMethod": int(float(d["LoginMethod"])),
         }
+
+
+class ConnectionType(StrEnum):
+    """Enum for connection types."""
+
+    WIRED = "wired"
+    WIRELESS = "wireless"
+
+
+@dataclass(kw_only=True)
+class AttachedDevice:
+    """Represents an attached device."""
+
+    ip_address: str
+    hostname: str | None = None
+    mac_address: str
+    connection_type: ConnectionType
+    link_speed: int | None = None
+    signal_strength: int
+    blocked: bool
+
+    @classmethod
+    def from_string(cls, source_string: str) -> AttachedDevice:
+        """Pre deserialize hook."""
+        parts = source_string.split(";")
+        ip_address = parts[1]
+        hostname = None
+        if parts[2] != "--":
+            hostname = parts[2]
+        mac_address = parts[3]
+        connection_type = ConnectionType(parts[4])
+        link_speed = None
+        if parts[5] != "":
+            link_speed = int(parts[5])
+        signal_strength = int(parts[6])
+        blocked = parts[7] != "Allow"
```

Comment:
If i am not mistaking, older models of routers respond with less "parts".
Therefore you may want to check the length of the parts:
https://github.com/MatMaul/pynetgear/blob/d9b34149b110cd91fbccc5fff7a285a1221af5dd/pynetgear/router.py#L649-L670